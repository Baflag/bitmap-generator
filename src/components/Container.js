import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import ColorsPalette from '../components/ColorsPalette/ColorsPalette';
import Board from '../components/Board';
import Canvas from '../components/Canvas';
import { mouseUp } from '../redux/actions';
import Carousel from '../components/Carousel';
import Modal from '../components/Modal';
import {
  AddAfterButton,
  AddBeforeButton,
  AddCopyButton,
  DownloadButton,
  ModalSwitch,
  ResetButton,
  SetSize
} from '../components/buttons';

const Container = ({ mouseUp }) => {
  return (
    <div>
      <div className="header">
        <div className="innerHeader">PIXEL ART</div>
      </div>
      <div>
        <div className="brd">
          <div
            className="ui raised segment"
            id="gridOne"
            onMouseLeave={mouseUp}
          >
            <Board />
            <SetSize />
          </div>
          <div id="gridTwo">
            <div id="gridThree">
              <ColorsPalette />
            </div>

            <div id="gridFour">
              <div className="grid-container">
                <div className="buttonMainContainer">
                  <div className="ui raised segment vertical buttons buttonColumnOne">
                    <DownloadButton />
                    <ResetButton />
                    <ModalSwitch />
                  </div>
                  <div className="ui raised segment vertical buttons buttonColumnTwo">
                    <AddBeforeButton />
                    <AddAfterButton />
                    <AddCopyButton />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <Carousel />
        </div>
        <Canvas />
        <Modal>TEXT</Modal>
      </div>
    </div>
  );
};

Container.propTypes = { mouseUp: PropTypes.func };

export default connect(
  null,
  { mouseUp }
)(Container);
