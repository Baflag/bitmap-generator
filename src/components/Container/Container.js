import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import ColorsPalette from '../ColorsPalette/ColorsPalette';
import Board from '../Board';
import Canvas from '../Canvas';
import Carousel from '../Carousel/Carousel';
import FrameAnimationModal from '../FrameAnimationModal';
import SetSize from '../buttons/SetSize';
import ReusableButton from '../buttons/ReusableButton';
import {
  mouseUp,
  addAfter,
  addBefore,
  addCopy,
  downloadBoard,
  modalSwitch,
  resetBoard
} from '../../redux/actions';

const Container = ({
  mouseUp,
  addAfter,
  addBefore,
  addCopy,
  downloadBoard,
  modalSwitch,
  resetBoard
}) => {
  return (
    <div className="container">
      <div className="background"></div>
      <div className="appContainer">
        <div className="appInnerContainer">
          <div className="header">
            <div className="innerHeader">PIXEL ART</div>
          </div>
          <div>
            <div className="brd">
              <div
                className="ui raised segment"
                id="gridOne"
                onMouseLeave={mouseUp}
              >
                <Board />
                <SetSize />
              </div>
              <div id="gridTwo">
                <div id="gridThree">
                  <ColorsPalette />
                </div>

                <div id="gridFour">
                  <div className="grid-container">
                    <div className="buttonMainContainer">
                      <div className="ui raised segment vertical buttons buttonColumnOne">
                        <ReusableButton
                          name={'download'}
                          onClick={downloadBoard}
                        />
                        <ReusableButton
                          name={'reset frame'}
                          onClick={resetBoard}
                        />
                        <ReusableButton
                          name={'show animation'}
                          onClick={modalSwitch}
                        />
                      </div>
                      <div className="ui raised segment vertical buttons buttonColumnTwo">
                        <ReusableButton name={'add after'} onClick={addAfter} />
                        <ReusableButton
                          name={'add before'}
                          onClick={addBefore}
                        />
                        <ReusableButton name={'copy frame'} onClick={addCopy} />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div className="carouselWrapper">
              <Carousel />
            </div>
            <Canvas />
            <FrameAnimationModal>TEXT</FrameAnimationModal>
          </div>
        </div>
      </div>
      <div className="background2"></div>
    </div>
  );
};

Container.propTypes = {
  mouseUp: PropTypes.func,
  addAfter: PropTypes.func,
  addBefore: PropTypes.func,
  addCopy: PropTypes.func,
  downloadBoard: PropTypes.func,
  modalSwitch: PropTypes.func,
  resetBoard: PropTypes.func
};

const mapDispatchToProps = {
  mouseUp: mouseUp,
  addAfter: addAfter,
  addBefore: addBefore,
  addCopy: addCopy,
  downloadBoard: downloadBoard,
  modalSwitch: modalSwitch,
  resetBoard: resetBoard
};

export default connect(
  null,
  mapDispatchToProps
)(Container);
