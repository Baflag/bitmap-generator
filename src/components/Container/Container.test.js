import React from 'react';
import Container from './Container';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import store from '../../store';

configure({ adapter: new Adapter() });

describe('<Container/>', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<Container store={store} />).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });
});
