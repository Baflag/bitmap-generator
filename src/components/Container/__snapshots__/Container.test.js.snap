// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container/> matches the snapshot 1`] = `
<div>
  <div
    className="header"
  >
    <div
      className="innerHeader"
    >
      PIXEL ART
    </div>
  </div>
  <div>
    <div
      className="brd"
    >
      <div
        className="ui raised segment"
        id="gridOne"
        onMouseLeave={[Function]}
      >
        <Connect(Board) />
        <Connect(SetSize) />
      </div>
      <div
        id="gridTwo"
      >
        <div
          id="gridThree"
        >
          <Connect(ColorsPalette) />
        </div>
        <div
          id="gridFour"
        >
          <div
            className="grid-container"
          >
            <div
              className="buttonMainContainer"
            >
              <div
                className="ui raised segment vertical buttons buttonColumnOne"
              >
                <ReusableButton
                  name="download"
                  onClick={[Function]}
                />
                <ReusableButton
                  name="reset frame"
                  onClick={[Function]}
                />
                <ReusableButton
                  name="show animation"
                  onClick={[Function]}
                />
              </div>
              <div
                className="ui raised segment vertical buttons buttonColumnTwo"
              >
                <ReusableButton
                  name="add after"
                  onClick={[Function]}
                />
                <ReusableButton
                  name="add before"
                  onClick={[Function]}
                />
                <ReusableButton
                  name="copy frame"
                  onClick={[Function]}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="carouselWrapper"
    >
      <Connect(Carousel) />
    </div>
    <Canvas />
    <Connect(Modal)>
      TEXT
    </Connect(Modal)>
  </div>
</div>
`;
